File: java/src/main/java/challenges/aoc2024/day1/Application.java
package challenges.aoc2024.day1;

import challenges.aoc2024.day1.service.DayOneSolver;
import challenges.aoc2024.day1.service.ListService;
import service.TaskSolver;

public class Application {
    public static void main(String[] args) {
        final TaskSolver solver = new TaskSolver(new DayOneSolver(new ListService()));
        solver.solveTasks();
    }
}

File: scala/src/main/scala/challenges/aoc2024/day1/Application.scala
package challenges.aoc2024.day1

import _root_.service.TaskSolver
import challenges.aoc2024.day1.service.{DayOneSolver, ListService}

object Application {
  def main(args: Array[String]): Unit = {
    val solver: TaskSolver = TaskSolver(DayOneSolver(ListService()))
    solver.solveTasks()
  }
}
